include/openmc/mesh.h

```diff
@@ -346,6 +347,11 @@ class UnstructuredMesh : public Mesh {
   void to_hdf5(hid_t group) const override;
 
   // Data members
+  xt::xtensor<int, 1> shape_; //!< Number of mesh elements in each dimension
```

Comment:
If you know this will always be an array of length 3, you should make it an `std::array<int, 3>` to avoid indirection.